### PR TITLE
Add info about branches in README.md

### PR DESCRIPTION
**Pull request by abellina**
This is a WIP branch to capture a suggestion by @revans2, made here: https://github.com/NVIDIA/spark-rapids/issues/300#issuecomment-651061157

I believe that front and center in the README would be good in this case, but can create a separate FAQ page.

### CODE REVIEW

**Comment by abellina** [comment]
Note I added it closest to the first `mvn` command, but we can move to another section too.

File: README.md [hunk 1]
```diff
@@ -37,6 +37,15 @@ may file one [here](https://github.com/NVIDIA/spark-rapids/issues/new/choose).
...
 ## Build
 
+There are two types of branches in this repository:
+
+* `branch-[version]`: are development branches which can change often. Note that we merge into 
+  the branch with the greatest version number, as that is our default branch.
+
+* `main`: is the branch with the latest released code, and the version tag (i.e. `v0.1.0`)
+  is held here. `main` will change with new releases, but otherwise it should not change with
+  every pull request merged, and should be a more stable branch.
```
Comment (kuhushukla) [inline README.md:47]: Nit: is a more stable branch , perhaps ? Instead if should be.

Comment (abellina) [reply README.md]: @kuhushukla, please take another look. I made it slightly different.

**Comment by jlowe** [comment]
build

**Pull request merged**